lib/bolt/cli.rb

```diff
@@ -197,6 +198,7 @@ def finalize_setup
       @parser_deprecations.each { |dep| Bolt::Logger.deprecation_warning(dep[:type], dep[:msg]) }
       config.deprecations.each { |dep| Bolt::Logger.deprecation_warning(dep[:type], dep[:msg]) }
 
+      FileUtils.rm_f(config.project.cache_file) if options[:clear_cache]
```

Comment:
```suggestion
      FileUtils.rm(config.project.cache_file) if options[:clear_cache] && config.project.cache_file.exist?
```

`FileUtils.rm_f` will suppress all StandardErrors, which we don't want to do. See Nick's comment regarding `FileUtils.rm_rf` here: https://github.com/puppetlabs/bolt/pull/1979#issuecomment-658942621